commit e962a0bbd96bbec72f934a0f8053bd5c26dc21f4
Author: Ray Strode <rstrode@redhat.com>
Date:   Wed Aug 27 16:01:28 2008 -0400

    respond w/ "" if user hits esc to password prompt
    
    This is so when we load the next plugin it won't
    get confused.

diff --git a/src/plugins/splash/details/plugin.c b/src/plugins/splash/details/plugin.c
index 50e13ab..3706e81 100644
--- a/src/plugins/splash/details/plugin.c
+++ b/src/plugins/splash/details/plugin.c
@@ -205,6 +205,12 @@ hide_splash_screen (ply_boot_splash_plugin_t *plugin,
 
   ply_trace ("hiding splash screen");
 
+  if (plugin->pending_password_answer != NULL)
+    {
+      ply_answer_with_string (plugin->pending_password_answer, "");
+      plugin->pending_password_answer = NULL;
+    }
+
   ply_window_set_keyboard_input_handler (window, NULL, NULL);
   ply_window_set_backspace_handler (window, NULL, NULL);
   ply_window_set_enter_handler (window, NULL, NULL);
diff --git a/src/plugins/splash/fade-in/plugin.c b/src/plugins/splash/fade-in/plugin.c
index 2641217..604c181 100644
--- a/src/plugins/splash/fade-in/plugin.c
+++ b/src/plugins/splash/fade-in/plugin.c
@@ -578,6 +578,12 @@ hide_splash_screen (ply_boot_splash_plugin_t *plugin,
 {
   assert (plugin != NULL);
 
+  if (plugin->pending_password_answer != NULL)
+    {
+      ply_answer_with_string (plugin->pending_password_answer, "");
+      plugin->pending_password_answer = NULL;
+    }
+
   ply_window_set_keyboard_input_handler (window, NULL, NULL);
   ply_window_set_backspace_handler (window, NULL, NULL);
   ply_window_set_enter_handler (window, NULL, NULL);
diff --git a/src/plugins/splash/spinfinity/plugin.c b/src/plugins/splash/spinfinity/plugin.c
index 44d66cf..cd9a85b 100644
--- a/src/plugins/splash/spinfinity/plugin.c
+++ b/src/plugins/splash/spinfinity/plugin.c
@@ -387,6 +387,12 @@ hide_splash_screen (ply_boot_splash_plugin_t *plugin,
 {
   assert (plugin != NULL);
 
+  if (plugin->pending_password_answer != NULL)
+    {
+      ply_answer_with_string (plugin->pending_password_answer, "");
+      plugin->pending_password_answer = NULL;
+    }
+
   ply_window_set_keyboard_input_handler (window, NULL, NULL);
   ply_window_set_backspace_handler (window, NULL, NULL);
   ply_window_set_enter_handler (window, NULL, NULL);
diff --git a/src/plugins/splash/text/plugin.c b/src/plugins/splash/text/plugin.c
index 11c2dc4..465d763 100644
--- a/src/plugins/splash/text/plugin.c
+++ b/src/plugins/splash/text/plugin.c
@@ -253,6 +253,12 @@ hide_splash_screen (ply_boot_splash_plugin_t *plugin,
 
   ply_trace ("hiding splash screen");
 
+  if (plugin->pending_password_answer != NULL)
+    {
+      ply_answer_with_string (plugin->pending_password_answer, "");
+      plugin->pending_password_answer = NULL;
+    }
+
   ply_window_set_keyboard_input_handler (window, NULL, NULL);
   ply_window_set_backspace_handler (window, NULL, NULL);
   ply_window_set_enter_handler (window, NULL, NULL);
